### PR TITLE
Change encoder input to be json's map[string]interface{}

### PR DESCRIPTION
**Pull request by guptasu**
Change encoder input to be json's map[string]interface{} instead of yaml's map[interface{}]interface{}

### CODE REVIEW

File: mixer/pkg/protobuf/yaml/dynamic/encoderUtil.go [hunk 1]
```diff
@@ -117,26 +116,6 @@ func protoKey(fieldNumber int, wireType uint64) []byte {
 	return buf
 }
 
-// Int64 typecasts input to int64 in a permissive way.
-func Int64(v interface{}) (int64, bool) {
```
Comment (mandarjog) [inline mixer/pkg/protobuf/yaml/dynamic/encoderUtil.go]: Does yaml.Int64 have the float64 case?

File: mixer/pkg/protobuf/yaml/castUtil.go [hunk 2]
```diff
@@ -45,6 +47,14 @@ func ToInt64(v interface{}) (int64, bool) {
 		return int64(c), true
 	case int64:
 		return c, true
+	case float64:
+		// json only supports 'number'
+		// therefore even integers are encoded as float
+		ii := int64(c)
```
Comment (mandarjog) [inline mixer/pkg/protobuf/yaml/castUtil.go]: This seems odd. We are trying to covert to float64 and the number is already that. I think this was added to the wrong place.

File: mixer/pkg/protobuf/yaml/dynamic/encoderUtil.go (hunk 1 above)
Comment (guptasu) [reply mixer/pkg/protobuf/yaml/dynamic/encoderUtil.go]: YES

File: mixer/pkg/protobuf/yaml/castUtil.go (hunk 2 above)
Comment (guptasu) [reply mixer/pkg/protobuf/yaml/castUtil.go]: We want the thing `c` to be a int64, but it comes to us as float64. This code is trying to cast it into int64 and see if we can do it without loosing data. This is similar to your code in `encoderUtil.go`, right ?

Comment (guptasu) [reply mixer/pkg/protobuf/yaml/castUtil.go]: Diff is hiding the location of the code. PTAL https://github.com/guptasu/istio/blob/5c200b33f1af2553f30d25c1ae78bb76390814ad/mixer/pkg/protobuf/yaml/castUtil.go

**Comment by guptasu** [comment]
@mandarjog re-lgtm please

**Comment by guptasu** [comment]
@mandarjog can you please LGTM and merge this PR ?

**Comment by guptasu** [comment]
@mandarjog re-lgtm. And whenever the submit bot is back (after code freeze), this PR will go in.

**Comment by guptasu** [comment]
@mandarjog re-lgtm.

**Pull request merged**